### DISCUSSION TITLE
What is an action and how is it difference from a controller?

### DISCUSSION THREAD

**Discussion by nbro** [symfony/symfony#54896] [Q&A]
This is just a terminology issue.

I understand the concept of a class with multiple methods, which can respond to certain HTTP requests. I also understand the concept of a controller in the MVC pattern. These exist in Java, Python and probably many other languages and frameworks. 

Now, the class that contains these methods, which are associated with routes or URLs, often has the suffix `Controller` and is called a _controller_ or _controller class_. However, I've also read that the methods in this class are called _controllers_. If we think of the MVC pattern, it makes sense to think of these methods as the actual controllers, so the controller class is just a class with a bunch of controllers, which should probably be related. I've also come across the term _action_. 

So, is an action just a controller (method)? Is there any difference between a controller (method) and an action?

**Comment by ghost** [comment]
The is no difference. These are aliases.
